modules/frontend/tracebyidsharding.go

```diff
@@ -97,20 +104,16 @@ func (s shardQuery) RoundTrip(r *http.Request) (*http.Response, error) {
 				overallError = rtErr
 			}
 
-			if shouldQuit(r.Context(), statusCode, overallError) {
-				return
-			}
-
-			// check http error
-			if rtErr != nil {
-				_ = level.Error(s.logger).Log("msg", "error querying proxy target", "url", innerR.RequestURI, "err", rtErr)
-				overallError = rtErr
+			// Check the context of the worker request
+			if shouldQuit(innerR.Context(), statusCode, overallError) {
 				return
 			}
 
-			// if the status code is anything but happy, save the error and pass it down the line
+			// if the status code is anything but happy, save the error and pass it
```

Comment:
why did we remove the `rtErr` check that was here?

Reply:
It looks like its unreachable to me.  The `shouldQuit` above would have caught it because `overallError` is checked and if non-nil above, then then we've returned.

Reply:
yup, good call!